triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -1592,7 +1592,7 @@ object Runner {
   }
 
   private[trigger] def numberOfInFlightCreateCommands(
-      commandId: String,
+      commandId: SValue,
```

Comment:
Maybe we could have have a tag or a wrapper for SValue that represent commandId.

Reply:
The obvious place to do this would be https://github.com/digital-asset/daml/blob/151e12b81ada89cb8c4aae2806521f024b85c6eb/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala#L383-L392

And the obvious definition would be something along the lines of:
```scala
sealed trait ContractId
def ContractId(id: String): SValue @@ ContractId = 
  Tag[SValue, ContractId](SRecord(???, ImmArray(Name.assertFromString("unpack")), ArrayList(SText(id))))
```

However, without knowing the trigger package ID, it's hard to eliminate the `???`.

So, unless I'm missing something, this line of attack doesn't appear to help?

If we keep `commandId` as having type `String` (at this definition site), then we still need the trigger package ID to generate the appropriate `SValue`.

With an instance of the trigger converter, we can simply use `fromCommandId`. Passing this converter object in (from the call site) is clearly possible (e.g. see https://github.com/digital-asset/daml/blob/5cf7613231f56ff3b152b76e09b4e2141c53f141/triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRuleSimulationLibTest.scala#L213-L218) but this appears to mix conversion concerns at a lower level?

Any alternative suggestions?

Reply:
Not sure we understand each other. Let park this subject for now.